python/services/github_service.py

```diff
@@ -65,6 +67,34 @@ def __init__(self, org_token: str, organisation_name: str) -> None:
         ), execute_timeout=60)
         self.organisation_name: str = organisation_name
 
+    def archive_inactive_repositories(self, repo_type_to_archive: str, last_active_cutoff_date: datetime,
```

Comment:
Just for awareness, this is the function I considered moving into a strategy pattern 👀

It contains the meat of our archiving Policy, and it would be nice to have the logic explicitly defined in a separate Class that we inject into services

But that seemed like a little much 😂